### PR TITLE
DO NOT MERGE: Try gcs-related things with travis

### PR DESCRIPTION
**Pull request by marcoderama**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,28 +2,47 @@ language: go
 sudo: required
 dist: xenial
 
+cache:
+  directories:
+  # We cache the Google Cloud SDK so we don't have to download it again on subsequent builds.
+  - $HOME/google-cloud-sdk
```
[the remark this answers is not in the document]
Comment (marcoderama) [reply .travis.yml]: I went with the install script based on Google's recommendation  (https://cloud.google.com/sdk/docs/downloads-interactive and https://cloud.google.com/solutions/continuous-delivery-with-travis-ci#understanding_the_travisyml_file).  It's also what the Helm guys do.  (See lines 42-45 of https://github.com/helm/helm/blob/master/.circleci/deploy.sh)  (There's also the Travis `deploy` stunt for non-PR mode in which case we'd leave the decision up to them...)

Easy enough to try the image though.  Does it matter that the `google/cloud-sdk` image is Debian-based?  (Or Alpine-based with the `slim` version.)

File: scripts/deploy-to-gcs.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -x
+set -eu
+
+# Do not prompt for user input when using any Google Cloud SDK methods.
+export CLOUDSDK_CORE_DISABLE_PROMPTS=1
+
+# The sdk install script errors if this directory already exists,
+# but Travis already created it when we marked it as cached.
```
[the remark this answers is not in the document]
Comment (marcoderama) [reply scripts/deploy-to-gcs.sh]: This tests whether the sdk is actually installed under a possibly empty `$HOME/google-cloud-sdk` dir.  Moot if we don't bother caching.

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcoderama) [reply .travis.yml]: I'll try the `google/cloud-sdk` image thing but here are some things to ponder:

- Turn out the Travis machine already includes the SDK.  We could just use that.  It's not most recent version though.  (I tried to `gcloud update` it but no go.
- 

https://cloud.google.com/sdk/install

**Comment by marcoderama** [comment]
Some experiments:

1. Using the cloud-sdk that comes pre-installed with the travis image:
   - Actually works, but...
   - Rather old version.  201.0.0 (although most recent for the OS installed)
   - Travis: https://travis-ci.org/lightbend/helm-charts/builds/428290044
   - Github: https://github.com/lightbend/helm-charts/commit/e7601bc716fb1f01a46281f6dcc24f357a49f981
2. Using the Google install script (as recommended by Google):
   - Works (including filtering docs folder) with newest version
   - Travis: https://travis-ci.org/lightbend/helm-charts/builds/428263580
   - Github: https://github.com/lightbend/helm-charts/commit/2e8efc9cec298d07237be471ef5f0c580ad4202f
3. Using apt-get (as per Google instructions at https://cloud.google.com/sdk/docs/downloads-apt-get)
   - Most recent version and can filter
   - Travis: https://travis-ci.org/lightbend/helm-charts/builds/428374193
   - Github: https://github.com/lightbend/helm-charts/commit/278213ec9161d194be8e2bdaf63b72d9f3e860b6
4. Using `google/cloud-sdk` docker image
   - Most recent version and can filter
   - Slowest option.  Most cumbersome code.
   - Travis: https://travis-ci.org/lightbend/helm-charts/builds/428432421 (with some debug output)
   - Github: https://github.com/lightbend/helm-charts/commit/f1df7eff6da37ccd4ac67cc2e69090071a05e8b5
5. Using built-in Travis `deploy` functionality
   - Very simple setup but not available on PR builds
   - Can't filter
   - Travis: https://travis-ci.org/lightbend/helm-charts/builds/428476900 (But deploy bit doesn't show up for some reason.  The files were all copied to GCS though.)
   - Github: https://github.com/lightbend/helm-charts/commit/59324d6ccf5d079f0ca3870de0c5eb398b9ab2bd

**Comment by marcoderama** [comment]
I would rank those choices in the order of 3, 2, 1, 4, 5.  Is there any compelling reason to go the docker route (choice 4)?  I don't see it, given that travis rebuilds the env every time anyway.  Choices 3 and 2 are much more straightforward imho.  I would go with choice 3 or 2.

**Comment by dbrinegar** [comment]
No Terraform option?  awww

just kidding!

Slightly more seriously:  Is this only for Travis?  It seems like deploying to gcs might be interesting dev step from time to time, might be nice to have a way that works reasonably well enough on a macbook too.  If that's a reasonable thing to consider in this choice...  a nice-to-have bonus point perhaps

**Comment by marcoderama** [comment]
@jsravn made the point that the docker version would be usable outside of travis, on a machine that didn't have the GC SDK installed.  For that reason I'll probably enshrine things using that method.

**Comment by marcoderama** [comment]
See the 'final' docker version of these changes in https://github.com/lightbend/helm-charts/pull/96

**Pull request closed without merging**